File: Leetcode/2294. Partition Array Such That Maximum Difference Is K.cpp
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        if (k == 0) return unique(nums.begin(), nums.end()) - nums.begin();
        int partitions = 0, Min = nums[0], Max = nums[0];
        for (int &num : nums) {
            Min = min(Min, num);
            Max = max(Max, num);
            if (Max - Min > k) {
                partitions++;
                Min = Max = num;
            }
        }
        return ++partitions;
    }
};

File: Leetcode/2294. Partition Array Such That Maximum Difference Is K.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        if k == 0: return len(set(nums))
        nums.sort()
        partitions, Min, Max = 0, nums[0], nums[0]
        for num in nums:
            Min = min(Min, num)
            Max = max(Max, num)
            if Max - Min > k:
                partitions += 1
                Min = Max = num
        return 1 + partitions